magento/pwa-studio
# [feature]: Cart Coupons

## Description

Adds the Coupon functionality to price adjustments.
Also updates price summary when a cart is added or removed.

![](https://i.gyazo.com/21f7f281da24085f0db81a545d7c038b.gif)

## Related Issue





Closes [PWA-75](https://jira.corp.magento.com/browse/PWA-75).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Add a product and go to `/cart`.
2. Add a coupon, `pwa1423` if you are on the AWS deployed app.
3. See price summary updated and coupon appears with "remove" button.
4. Click remove
5. See price summary updated and coupon is removed.

**Additional Steps:**

The `Field` component did not provide adequate spacing for a scenario where two fields are aligned next to each other in a grid and one field has a label (or error message and the other doesn't).

In order to remedy this I added a `min-height` value to the field label style and made field components align their content to the start rather than be stretched by default.

Please ensure that `Field`s across Venia still appear correctly and that there are no regressions.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
